inst/scripts/query-pubmed-grants.R

```diff
@@ -221,104 +223,113 @@ pmids_df <- pmids_df %>% group_by(pmid) %>% reframe(
-
-## ----columns--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-dat <- set_up_multiannotations(dat, "grant")
-dat <- set_up_multiannotations(dat, "Program")
-dat <- set_up_multiannotations(dat, "Authors")
-
-
-## -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-store_as_annotations <- function(parent, list) {
-  entity <- purrr::map(
-    list,
-    ~ synapseclient$File(
-      path = glue::glue("http://doi.org/{.$DOI}"),
-      name = .$entity_name,
-      parent = parent,
-      synapseStore = FALSE,
-      annotations = .
+  # Remove common, unallowed characters from entity name; includes hacky_cleaning
+  dat$entity_name <- remove_unacceptable_characters(dat$entity_name)
+
+  ## ----columns--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+  dat <- set_up_multiannotations(dat, "grant")
+  dat <- set_up_multiannotations(dat, "Program")
+  dat <- set_up_multiannotations(dat, "Authors")
+
+  ## -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+  store_as_annotations <- function(parent, dat_list) {
+    entity <- purrr::map(
+      dat_list,
+      ~ synapseclient$File(
+        path = glue::glue("http://doi.org/{.$DOI}"),
+        name = .$entity_name,
+        parent = parent,
+        synapseStore = FALSE,
+        annotations = .
+      )
     )
-  )
-  # entity
-  purrr::map(entity, ~ syn$store(., forceVersion = FALSE))
+    # entity
+    purrr::map(entity, ~ syn$store(., forceVersion = FALSE))
+  }
+
+  ## ----store, message=FALSE, echo=FALSE-------------------------------------------------------------------------------------------------------------------------------------------------
+  # parent = "syn51317180" # ELITE publications folder
+  dat_list <- purrr::transpose(dat)
+  # another eternity
+  store_as_annotations(parent = sid_pub_folder, dat_list=dat_list)
 }
 
-## ----store, message=FALSE, echo=FALSE-------------------------------------------------------------------------------------------------------------------------------------------------
-# parent = "syn51317180" # ELITE publications folder
-dat_list <- purrr::transpose(dat)
+# For some reason there is a warning about leaded semaphore that is
+# leading to the process to never finish.
+reticulate::py_run_string("
+import warnings
 
-# another eternity
-store_as_annotations(parent = sid_pub_folder, dat_list)
+# Suppress specific multiprocessing warnings
+warnings.filterwarnings('ignore', message='resource_tracker: There appear to be .*')
```

Comment:
The script seems to hang for some reason due to this multiprocessing error, so I added code to specifically ignore it.